### ISSUE TITLE
SRT: Service stopped by assert fail.

### ISSUE THREAD

**Issue by nrgcreative** [ossrs/srs#3316]
**SRS Service stopped. Is this about SRT streaming interruption sometimes? **

1. SRS Version: `SRS/6.0.4`

1. SRS Log:

```
[2022-12-15 11:00:20.458][ERROR][2203786][aa2qr2p3][4] srt serve error code=6002(SrtTimeout)(SRT connection is timeout) : srt: recvmsg : wait readable : srt socket 271789055 timeout
thread [2203786][aa2qr2p3]: do_publishing() [./src/app/srs_app_srt_conn.cpp:426][errno=62]
thread [2203786][aa2qr2p3]: recvmsg() [./src/protocol/srs_protocol_srt.cpp:801][errno=62]
thread [2203786][aa2qr2p3]: wait_readable() [./src/protocol/srs_protocol_srt.cpp:861][errno=62](Interrupted system call)
[2022-12-15 11:07:39.097][ERROR][2203786][2g0g9j10][0] backtrace 20 frames of ./objs/srs SRS/6.0.4(Bee)
[2022-12-15 11:07:39.129][ERROR][2203786][2g0g9j10][0] #0 0x6cd40 0 /lib/x86_64-linux-gnu/libasan.so.5(+0x6cd40) [0x7f98a055bd40]
[2022-12-15 11:07:39.152][ERROR][2203786][2g0g9j10][0] #1 0x3b65a8 1 0x00000000003b65a7: SrsCplxError::srs_assert(bool) at srs_kernel_error.cpp:376
[2022-12-15 11:07:39.178][ERROR][2203786][2g0g9j10][0] #2 0x5417e8 1 0x00000000005417e7: SrsRawH264Stream::is_sps(char*, int) at srs_protocol_raw_avc.cpp:63
[2022-12-15 11:07:39.208][ERROR][2203786][2g0g9j10][0] #3 0x918983 1 0x0000000000918982: SrsRtmpFromSrtBridge::on_ts_video(SrsTsMessage*, SrsBuffer*) at srs_app_srt_source.cpp:382
[2022-12-15 11:07:39.230][ERROR][2203786][2g0g9j10][0] #4 0x9183a0 1 0x000000000091839f: SrsRtmpFromSrtBridge::on_ts_message(SrsTsMessage*) at srs_app_srt_source.cpp:350
[2022-12-15 11:07:39.257][ERROR][2203786][2g0g9j10][0] #5 0x3ecf1f 1 0x00000000003ecf1e: SrsTsContext::decode(SrsBuffer*, ISrsTsHandler*) at srs_kernel_ts.cpp:272
[2022-12-15 11:07:39.287][ERROR][2203786][2g0g9j10][0] #6 0x91784a 1 0x0000000000917849: SrsRtmpFromSrtBridge::on_packet(SrsSrtPacket*) at srs_app_srt_source.cpp:299 (discriminator 3)
[2022-12-15 11:07:39.309][ERROR][2203786][2g0g9j10][0] #7 0x91d090 1 0x000000000091d08f: SrsSrtSource::on_packet(SrsSrtPacket*) at srs_app_srt_source.cpp:797
[2022-12-15 11:07:39.336][ERROR][2203786][2g0g9j10][0] #8 0x90e6b8 1 0x000000000090e6b7: SrsMpegtsSrtConn::on_srt_packet(char*, int) at srs_app_srt_conn.cpp:540
[2022-12-15 11:07:39.361][ERROR][2203786][2g0g9j10][0] #9 0x90ca43 1 0x000000000090ca42: SrsMpegtsSrtConn::do_publishing() at srs_app_srt_conn.cpp:431
[2022-12-15 11:07:39.388][ERROR][2203786][2g0g9j10][0] #10 0x90abb6 1 0x000000000090abb5: SrsMpegtsSrtConn::publishing() at srs_app_srt_conn.cpp:319
[2022-12-15 11:07:39.410][ERROR][2203786][2g0g9j10][0] #11 0x90a647 1 0x000000000090a646: SrsMpegtsSrtConn::do_cycle() at srs_app_srt_conn.cpp:293
[2022-12-15 11:07:39.437][ERROR][2203786][2g0g9j10][0] #12 0x9090cd 1 0x00000000009090cc: SrsMpegtsSrtConn::cycle() at srs_app_srt_conn.cpp:226
[2022-12-15 11:07:39.463][ERROR][2203786][2g0g9j10][0] #13 0x68eb02 1 0x000000000068eb01: SrsFastCoroutine::cycle() at srs_app_st.cpp:285
[2022-12-15 11:07:39.490][ERROR][2203786][2g0g9j10][0] #14 0x68ec52 1 0x000000000068ec51: SrsFastCoroutine::pfn(void*) at srs_app_st.cpp:300
[2022-12-15 11:07:39.513][ERROR][2203786][2g0g9j10][0] #15 0xa2fec6 1 0x0000000000a2fec5: _st_thread_main at sched.c:380
[2022-12-15 11:07:39.543][ERROR][2203786][2g0g9j10][0] #16 0xa307ec 1 0x0000000000a307eb: st_thread_create at sched.c:666 (discriminator 1)
[2022-12-15 11:07:39.568][ERROR][2203786][2g0g9j10][0] #17 0x5c19af 1 0x00000000005c19ae: SrsSrtPoller::mod_socket(SrsSrtSocket*) at srs_protocol_srt.cpp:656
[2022-12-15 11:07:39.594][ERROR][2203786][2g0g9j10][0] #18 0xa2f694 1 0x0000000000a2f693: _st_vp_schedule at sched.c:162
[2022-12-15 11:07:39.624][ERROR][2203786][2g0g9j10][0] #19 0xdbf640 0 ./objs/srs(+0xdbf640) [0x55996f3b6640]
```

1. SRS Config:

```

listen              5035;
max_connections     100;
srs_log_level       error;
srs_log_file        ./objs/srs.log;
daemon              on;

http_api {
    enabled         on;
    listen          1985;
}
http_server {
    enabled         on;
    listen          8080;
    dir             ./objs/nginx/html;
}


srt_server {
    enabled on;
    listen 20000;
    connect_timeout 4000;
    peerlatency 0;
    recvlatency 0;
    latency 250;
    default_app livetv;
    tlpktdrop off;
    tsbpdmode off;
}


vhost __defaultVhost__ {

    tcp_nodelay     on;
    min_latency     on;
    gop_cache       on;
   
    srt {
        enabled     on;
     }

    hls {
        enabled off;
    }

    http_remux {
        enabled     on;
        mount       [vhost]/[app]/[stream].flv;
    }

    publish {
         mr          off;
        mr_latency  350;
    }

}

```

**Replay:**

**> Please describe how to replay the bug? (重现Bug的步骤)**

1. `SRS Server Down`

**Expect (Expected Behavior)**

> I expect it will work stable..

`TRANS_BY_GPT3`

**Comment by winlinvip** [comment]
Thank you for your feedback, with detail logs. We will fix it ASAP.

**Comment by winlinvip** [comment]
Fixed in v5.0.112 and v6.0.6
